tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -1119,6 +1119,40 @@ void test_object_recycle(void)
 	zassert_true(perms_count == 1, "invalid number of thread permissions");
 }
 
+#if !defined(CONFIG_ARC) /* FIXME: #17590 */
```

Comment:
these functions aren't static so strictly speaking you don't need to ifdef around !CONFIG_ARC

Reply:
sure